src/board.rs

```diff
@@ -91,6 +95,7 @@ impl Board {
             }
         }
         self.squares[to.y][to.x] = piece;
+        self.squares[from.y][from.x] = None;
```

Comment:
why are we doing this? the code before does the exact same thing but it's shorter. `take()` leaves `None` inside

Reply:
It was just doing `take` on a copy, it wasn't changing the one in `squares`.